.github/workflows/pr-deploy.yaml

```diff
@@ -0,0 +1,191 @@
+      CODER_BASE_IMAGE_TAG: ${{ steps.set_tags.outputs.CODER_BASE_IMAGE_TAG }}
+      CODER_IMAGE_TAG: ${{ steps.set_tags.outputs.CODER_IMAGE_TAG }}
+
+    runs-on: "ubuntu-latest"
+    steps:
+      - name: Get PR number
+        id: pr_number
+        run: |
+          set -euxo pipefail
+          if [[ ${{ github.event_name }} == 'workflow_dispatch' ]]; then
+            PR_NUMBER=${{ github.event.inputs.pr_number }}
+          else
+            PR_NUMBER=${{ github.event.issue.number }}
+          fi
+          echo $PR_NUMBER
+          echo "PR_NUMBER=$PR_NUMBER" >> $GITHUB_OUTPUT
+
+      - name: Set required tags
+        id: set_tags
+        run: |
+          set -euxo pipefail
+          echo "CODER_BASE_IMAGE_TAG=$CODER_BASE_IMAGE_TAG" >> $GITHUB_OUTPUT
+          echo "CODER_IMAGE_TAG=$CODER_IMAGE_TAG" >> $GITHUB_OUTPUT
+        env:
+          CODER_BASE_IMAGE_TAG: ghcr.io/coder/coder-preview-base:pr${{ steps.pr_number.outputs.PR_NUMBER }}
+          CODER_IMAGE_TAG: ghcr.io/coder/coder-preview:pr${{ steps.pr_number.outputs.PR_NUMBER }}
+
+      - name: Find Comment
+        uses: peter-evans/find-comment@v2
+        id: fc
+        with:
+          issue-number: ${{ steps.pr_number.outputs.PR_NUMBER }}
+          comment-author: "github-actions[bot]"
+          body-includes: This deployment will be deleted when the PR is closed
+
+          # Comment on the PR
+      - name: Comment on PR
+        id: comment_id
+        uses: peter-evans/create-or-update-comment@v3
+        with:
+          comment-id: ${{ steps.fc.outputs.comment-id }}
+          issue-number: ${{ steps.pr_number.outputs.PR_NUMBER }}
+          edit-mode: replace
+          body: |
+            :rocket: Deploying PR ${{ steps.pr_number.outputs.PR_NUMBER }}
+            :warning: This deployment will be deleted when the PR is closed.
+
+  build:
+    needs: pr_commented
+    runs-on: ${{ github.repository_owner == 'coder' && 'buildjet-8vcpu-ubuntu-2204' || 'ubuntu-latest' }}
+    env:
+      # Necessary for Docker manifest
+      DOCKER_CLI_EXPERIMENTAL: "enabled"
+      CODER_BASE_IMAGE_TAG: ${{ needs.pr_commented.outputs.coder_base_image_tag }}
+      CODER_IMAGE_TAG: ${{ needs.pr_commented.outputs.coder_image_tag }}
+      PR_NUMBER: ${{ needs.pr_commented.outputs.pr_number }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - uses: ./.github/actions/setup-go
+
+      - uses: ./.github/actions/setup-node
+
+      - name: Install sqlc
+        run: |
+          curl -sSL https://github.com/kyleconroy/sqlc/releases/download/v1.18.0/sqlc_1.18.0_linux_amd64.tar.gz | sudo tar -C /usr/bin -xz sqlc
+
+      - name: GHCR Login
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Create empty base-build-context directory
+        run: mkdir base-build-context
```

Comment:
Our base image hardly changes. How difficult would it be to avoid building the base image, and only build if changes are detected in `./scripts/Dockerfile.base`. I'm assuming the benefit would be faster builds/preview-time. 

I know there are other optimizations we can make as well. Since this is intended to be a "quick preview environment" to avoid manually building/deploying, we should definitely investigate any substantial improvements to build time.

Reply:
I have 2 ideas,
1. Use the alreday existing base image from last release `ghcr.io/coder/coder-base:latest`
2. 1 but build a new base-image if teh PR targets the `Dockerfile.base` file.

2nd would be ideal.

Reply:
I did implement the 2nd option, And it seems to work. Thanks for the feedback